test/DotNetEnv.Tests/ParserTests.cs

```diff
@@ -361,6 +392,23 @@ public void ParseAssignment ()
             Assert.Throws<ParseException>(() => Parsers.Assignment.End().Parse("EV_DNE='"));
             Assert.Throws<ParseException>(() => Parsers.Assignment.End().Parse("EV_DNE=0\n1"));
 
+            testParse("EV_DNE", "", "EV_DNE=");
+            testParse("EV_DNE", "a b c", "EV_DNE=a b c # comment");
+            testParse("EV_DNE", "test", "EV_DNE= test  #a'bc allow singleQuotes in comment");
+            testParse("EV_DNE", "test", "EV_DNE= test  #a\"bc allow doubleQuotes in comment");
+            testParse("EV_DNE", "test", "EV_DNE= test #a$bc allow dollarSign in comment");
+
+            testParse("EV_DNE", "http://www.google.com/#anchor", "EV_DNE=http://www.google.com/#anchor");
+            testParse("EV_DNE", "", "EV_DNE= #no value just comment");
+            testParse("EV_DNE", "", "EV_DNE=#no value just comment");
+            testParse("EV_DNE", "a'b''c", "EV_DNE=a'b''c #allow inline singleQuotes in unquoted values");
```

Comment:
I remain uncertain if this is truly desirable, but whatevs, it's easy to support and doesn't seem like a problem -- especially given the below, where quoted values all still seem to work as desired

Reply:
It would be great to have a standard to stick to, but there's no such standard for env afaik.

In my opinion there is no reason to disallow any character within unquoted values except for leading special-chars (quotation-chars and comment-begin)
Since we even allow inline quotation-char if it's not preceeded with tab or whitespace, I guess we should allow quotation-chars too.